### PR TITLE
use sync method for mapPage

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Here is the sync approach to mapPage. I would make a transform from PagedIterable to another PagedIterable directly on its stream and iterable, skipping the underlying PagedFlux.

This way I will not need to change any code in azure-core. I only need to put this 100+ lines of code to each mgmt SDK (100+ of them).

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@srnagar @anuchandy 

Here is the sync approach. I would make a transform from `PagedIterable` to another `PagedIterable` directly, skipping the underlying `PagedFlux` (however the constructor taking a `PagedFlux` is a pain in the ###).

This way I will not need to change any code in azure-core. I only need to put this 100+ lines of code to each mgmt SDK (100+).

Would you give a quick review on whether this approach is kind of correct to you? (you do not need to be very detailed here since it will not get into azure-core)

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 1]
```diff
@@ -49,6 +52,123 @@ public PagedIterable(PagedFlux<T> pagedFlux) {
...
+    }
+
+    private static final class PagedIterableImpl<T, S> extends PagedIterable<S> {
+
+        private final PagedIterable<T> source;
+        private final Function<T, S> mapper;
+        private final Function<PagedResponse<T>, PagedResponse<S>> pageMapper;
+
+        private PagedIterableImpl(PagedIterable<T> sourcePageIterable, Function<T, S> mapper) {
+            super(new PagedFlux<S>(Mono::empty));
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java:108]: this is a pain

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java:108]: We could move the other paging iterables into `implementation` to re-use the code and add any additional hooks to allow mapping of pages to be performed

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java:108]: It there is no plan to deprecate this API (as with `PagedFlux.mapPage`), and if the implementation is done correctly, I believe this solution would be superior than the `PagedFlux.mapPage` approach.

Let's take time for review and discussion. Meanwhile I will switch our usage to code like this, so no urgency.

**Comment by weidongxu-microsoft** [comment]
OK. I've already put this in generated code.

The problem in azure-core still existing, but at least for now mgmt SDK will not use this problematic implementation.

**Pull request closed without merging**